addons/web/static/src/views/form/form_controller.scss

```diff
@@ -981,20 +982,22 @@
     .o_form_view {
         min-height: auto;
 
-        --formView-sheetBg-padding-top: #{map-get($spacers, 2)};
-        --formView-sheetBg-padding-x: #{$modal-inner-padding};
+        .o_form_sheet_bg {
+            --formView-sheetBg-padding-top: 0;
+            --formView-sheetBg-padding-x: 0;
+        }
     }
 
     .o_form_sheet {
         --formView-sheet-margin-x: 0;
-        --formView-sheet-padding-y: #{map-get($spacers, 2)};
-        --formView-sheet-padding-y-md: #{map-get($spacers, 2)};
-        --formView-sheet-padding-y-lg: #{map-get($spacers, 2)};
-        --formView-sheet-padding-y-xxl: #{map-get($spacers, 2)};
-        --formView-sheet-padding-x: 0;
-        --formView-sheet-padding-x-md: 0;
-        --formView-sheet-padding-x-lg: 0;
-        --formView-sheet-padding-x-xxl: 0;
+        --formView-sheet-padding-y: #{map-get($spacers, 3)};
+        --formView-sheet-padding-y-md: #{map-get($spacers, 3)};
+        --formView-sheet-padding-y-lg: #{map-get($spacers, 3)};
+        --formView-sheet-padding-y-xxl: #{map-get($spacers, 3)};
+        --formView-sheet-padding-x: #{map-get($spacers, 3)};
+        --formView-sheet-padding-x-md: #{map-get($spacers, 3)};;
+        --formView-sheet-padding-x-lg: #{map-get($spacers, 3)};;
```

Comment:
Kinda similar to the comment above, if you go to Approvals > Time Off and create a new Time Off, we have a `padding` which prevents the grey background to stick to the bottom of the form view.

I'm not sure you can find a perfect solution but maybe it's worth having a look?

**saas-16.3 :**
![image](https://github.com/odoo/odoo/assets/128030743/c75c4154-fe60-49ef-b6f1-c3f9f5ef9c7e)

**saas-16.3-modal-fixes-edi :**
![image](https://github.com/odoo/odoo/assets/128030743/9edb883d-affa-402d-8f03-9a59430a10b5)